src/inbound.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-env serviceworker */
+import { Router } from 'itty-router'
+import { JSON404, JSON500, JSONResponse } from './utils/http/responses.js'
+import { publishRecord } from './publish.js'
+
+// We might use a different router, depending on how this is hosted
+const router = Router()
+router.options('*', corsOptions)
+router.post('/broadcast', broadcast)
+router.get('/', siteRoot)
+router.all('*', () => JSON404())
+
+/**
+ * Receive a request containing an IPNS record and publish it to the DHT.
+ * @returns {JSONResponse}
+ */
+function broadcast (request, env, ctx) {
+  // TODO:
+  // 1. Validate record - can we borrow code from the w3name repo? See w3name/#9
```

Comment:
I think we could import the method from here: https://github.com/web3-storage/w3name/pull/11/files#diff-c18413c84e86453cd731abf8a6867ef9e5783e8a904aa5b2faeda3168bae2cc1

Reply:
Given that this endpoint is now private to w3name and w3name does its own validation, we decided not to re-check the record here.